model/job.go

```diff
@@ -225,6 +234,123 @@ func (j *Job) loadJobSpec() (err error) {
 	return nil
 }
 
+// MergeSpec is used to merge temporary spec patches into each job. otherJob should only be
+// the hcf/patch-properties job.
+func (j *Job) MergeSpec(otherJob *Job) {
+	// Ignore otherJob.Name, otherJob.Description
+	if otherJob.Packages != nil {
+		j.Packages = append(j.Packages, otherJob.Packages...)
+	}
+	// Skip templates
+	// Question for Vlad: how are the patch-properties templates used?
+	if otherJob.Properties != nil {
+		j.Properties = append(j.Properties, otherJob.Properties...)
+	}
+}
+
+// WriteConfigs merges the job's spec with the opinions and writes out json-encoding to the specified path.
+func (j *Job) WriteConfigs(role *Role, outputPath, lightOpinionsPath, darkOpinionsPath string) (err error) {
+	config, err := initializeConfigJSON()
+	if err != nil {
+		return err
+	}
+
+	config["job"].(map[string]interface{})["name"] = role.Name
+
+	var templates []map[string]string
+	for _, roleJob := range role.Jobs {
+		templates = append(templates, map[string]string{"name": roleJob.Name})
+	}
+	config["job"].(map[string]interface{})["templates"] = templates
+
+	opinions, err := newOpinions(lightOpinionsPath, darkOpinionsPath)
```

Comment:
I wish we passed in the opinion directly, but I guess that's currently not available to the caller :|